tests/test_internals/test_si.py

```diff
@@ -576,5 +585,68 @@ def test_group_set_custom_description(self):
                     assert nwb.electrodes["group"][i].description == "M1 description"
 
 
+class TestSpikeInterfaceRecorders(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+
+        cls.test_dir = tempfile.mkdtemp()
+        cls.path1 = cls.test_dir + "/test_electrodes1.nwb"
+
+        cls.nwbfile1 = NWBFile("session_description1", "file_id1", datetime.now())
+
+        cls.RX1 = generate_recording(num_channels=4, durations=[3])
+        cls.metadata_list = [dict(Ecephys={i: dict(name=i, description="desc")}) for i in ["es1", "es2"]]
+
+        channel_ids = cls.RX1.get_channel_ids()
+        num_channels = cls.RX1.get_num_channels()
+        non_int_channel_ids1 = ["a", "b", "c", "d"]
+        non_int_channel_ids2 = ["c", "d", "e", "f"]
+
+        cls.RX_non_int_channels1 = cls.RX1.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids1
+        )
+        cls.RX_non_int_channels2 = cls.RX1.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids2
+        )
+
+        # Property for the first recorder
+        values_before_rewrite = ["value_before_rewrite" for _ in range(num_channels)]
+        cls.RX_non_int_channels1.set_property("property", values=values_before_rewrite)
+
+        # Property in the second recorder that extends the one above
+        values_after_rewrite = ["value_after_rewrite" for _ in range(num_channels)]
+        cls.RX_non_int_channels2.set_property("property", values=values_after_rewrite)
+
+        # Add a property only available in the second recorder
+        property_value = ["second_recorder_property" for _ in range(num_channels)]
+        cls.RX_non_int_channels2.set_property("property2", values=property_value)
+
+        # Write the two electrodes to a file
+        add_electrodes(recording=cls.RX_non_int_channels1, nwbfile=cls.nwbfile1)
+        add_electrodes(recording=cls.RX_non_int_channels2, nwbfile=cls.nwbfile1)
+
+    @classmethod
+    def tearDownClass(cls):
+        del cls.RX1, cls.RX_non_int_channels1, cls.RX_non_int_channels2
```

Comment:
Can you look into if this manual deletion of variables is necessary? I'm pretty sure it should happen automatically, whereas the temporary folder directory cleanup is something we definitely are responsible for.

Reply:
It is actually not necessary, I removed it.